TransformationSystem/Agent/ValidateOutputDataAgent.py

```diff
@@ -6,171 +6,170 @@
 import re
 import ast
 
-from DIRAC                                                     import S_OK, S_ERROR, gLogger
-from DIRAC.Core.Base.AgentModule                               import AgentModule
-from DIRAC.ConfigurationSystem.Client.Helpers.Operations       import Operations
-from DIRAC.DataManagementSystem.Client.DataIntegrityClient     import DataIntegrityClient
-from DIRAC.Resources.Catalog.FileCatalog                       import FileCatalog
-from DIRAC.Resources.Catalog.FileCatalogClient                 import FileCatalogClient
-from DIRAC.TransformationSystem.Client.TransformationClient    import TransformationClient
-from DIRAC.DataManagementSystem.Client.ConsistencyInspector    import ConsistencyInspector
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+from DIRAC.DataManagementSystem.Client.DataIntegrityClient import DataIntegrityClient
+from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+from DIRAC.Resources.Catalog.FileCatalogClient import FileCatalogClient
+from DIRAC.TransformationSystem.Client.TransformationClient import TransformationClient
+from DIRAC.DataManagementSystem.Client.ConsistencyInspector import ConsistencyInspector
 
 AGENT_NAME = 'Transformation/ValidateOutputDataAgent'
 
-class ValidateOutputDataAgent( AgentModule ):
 
-  def __init__( self, *args, **kwargs ):
+class ValidateOutputDataAgent(AgentModule):
+
+  def __init__(self, *args, **kwargs):
     """ c'tor
     """
-    AgentModule.__init__( self, *args, **kwargs )
-    
+    AgentModule.__init__(self, *args, **kwargs)
+
     self.consistencyInspector = ConsistencyInspector()
     self.integrityClient = DataIntegrityClient()
     self.fc = FileCatalog()
     self.transClient = TransformationClient()
     self.fileCatalogClient = FileCatalogClient()
 
-    agentTSTypes = self.am_getOption( 'TransformationTypes', [] )
+    agentTSTypes = self.am_getOption('TransformationTypes', [])
     if agentTSTypes:
       self.transformationTypes = agentTSTypes
     else:
-      self.transformationTypes = Operations().getValue( 'Transformations/DataProcessing', ['MCSimulation', 'Merge'] )
+      self.transformationTypes = Operations().getValue('Transformations/DataProcessing', ['MCSimulation', 'Merge'])
 
-    self.directoryLocations = sorted( self.am_getOption( 'DirectoryLocations', ['TransformationDB',
-                                                                                  'MetadataCatalog'] ) )
-    self.activeStorages = sorted( self.am_getOption( 'ActiveSEs', [] ) )
```

Comment:
Shouldn't this option stay ? What is it actually used for ? As a starting point to crawl the data ? or just to make sure we only remove from the concerned storage (and avoid for example SWTEST and the rest) ?

Reply:
This is not removing anything, but used to check the data is written on the SE. But this is done by directory listing, and we don't want it. Plus it's very fragile.